### PR TITLE
Phalcon 3.4.x on PHP 8

### PR DESCRIPTION
**Pull request by Bellardia**
Introduces the required Packer changes to build PHP 8 (and 7.4) and MySql 5.7 on Ubuntu Bionic and Ubuntu Focal.

We are limited to running Ubuntu Bionic in AWS OpsWorks, so that will continue to be our base environment for the time being, however I ported everything over to Ubuntu Focal (20.04) as well and it's possible to use such images everywhere except in AWS OpsWorks at the moment (in theory the work in raw EC2, it's namely the `OpsWorks Agent` does not support anything `> 18.04` so instances cannot be managed by OpsWorks).

### CODE REVIEW

File: base/recipes/mysql.rb [hunk 1]
```diff
@@ -28,11 +35,32 @@
...
+["mysql-client-core", "mysql-client", "mysql-server-core", "mysql-server"].each do |pkg|
+  execute "DEBIAN_FRONTEND=noninteractive dpkg -i #{cookbook_files}/#{pkg}-#{node["mysql"]["version"]}_#{arch}.deb || true" do
+    not_if "dpkg -S #{pkg}- | grep '^#{pkg}-#{node["mysql"]["version"].to_i}'"
+    notifies :run, "execute[mysql-configure]", :before if pkg == "mysql-server"
+    notifies :edit, "replace_or_add[mysql-dpkg-configure]", :immediately if pkg == "mysql-server"
+    notifies :stop, "service[mysql]", :immediately if pkg == "mysql-server"
+    notifies :disable, "service[mysql]", :immediately if pkg == "mysql-server"
+  end
+end
+
```
Comment (ahosking) [inline base/recipes/mysql.rb]: @Bellardia is this the only reliable way to pin our versions and keep mysql humming? 
Storing the binaries seems like a bummer, but, necessary.
I recall from the last time I worked on this that getting the repos from other distribution versions did cause more problems than it solved.

File: configure/templates/default/my.cnf.erb [hunk 2]
```diff
@@ -170,9 +169,9 @@ sort_buffer_size                = 1M    # UPD
 # to be overriden (also see comment next to open_files_limit).
 # E.g. for MySQL 5.7, please check: https://dev.mysql.com/doc/refman/5.7/en/using-systemd.html
 # and for MariaDB check: https://mariadb.com/kb/en/library/systemd/
-table_definition_cache          = 50000 # UPD
-table_open_cache                = 50000 # UPD
-open_files_limit                = 100000 # UPD - This can be 2x to 3x the table_open_cache value or match the system's
+table_definition_cache          = 32713 # UPD
+table_open_cache                = 32713 # UPD
+open_files_limit                = 65565 # UPD - This can be 2x to 3x the table_open_cache value or match the system's
```
Comment (ahosking) [inline configure/templates/default/my.cnf.erb:174]: What is the significance of these new, lower values?

**Review by ahosking — APPROVED** [review]

File: base/recipes/mysql.rb (hunk 1 above)
Comment (Bellardia) [reply base/recipes/mysql.rb]: yes, the dependencies can't be built on 18.04 due to other missing dependencies. I need to use pre-built versions from other sources

there's a way to do it by installing through mysql 8 but it puts the system in a worse state overall with different dep issues

File: configure/templates/default/my.cnf.erb (hunk 2 above)
Comment (Bellardia) [reply configure/templates/default/my.cnf.erb:174]: pretty sure they're just under the default open files limit on docker which is less than they were on vagrant. 
as to what level of the stack is enforcing them im not sure, but the numbers are high enough that it's not especially consequential

**Review by ahosking — APPROVED** [review]

**Pull request merged**